### PR TITLE
Add new attention and normalization kernels

### PR DESCRIPTION
**Pull request by Svoloch2940194**
- **New optional `attentions` module** – added to the build system with the `BUILD_ATTENTIONS_MODULE` flag so you can choose whether to compile and include these kernels.  
  Enabled by default. When building, the flag **determines whether the module is built or not**.
  - `laser_attention` – LASER attention kernel
  - `block_sparse_attention` – block‑sparse attention 
  - `sparse_block_estimate` – for block‑sparse attention 
  - `rainfusionattention` – RainFusion attention kernel
  - `layernorm` – optimized layer normalization kernel
- **Build system updates**
  - Separate compilation step via `build_attentions_kernels` (sources in `csrc/attentions/build`)
  - Packaging step via `make_attentions_package` produces a standalone `attentions*.whl` in `OUTPUT_DIR`

### How to build
```bash
# Build the library with attention kernels enabled (default)
bash build.sh
```
After a successful build, you will find the wheel inside the output directory.

The kernels are registered as standard PyTorch custom ops under torch.ops.attentions
### How to use 
```python
import attentions
import torch 
# next use torch.ops.attentions
```

The tables with measurements can be found in this PR: https://github.com/sgl-project/sglang/pull/23482

### CODE REVIEW

File: python/attentions/attentions/plugin/__init__.py
Comment (ping1jing2) [inline python/attentions/attentions/plugin/__init__.py:1]: why do we need such dir?

File: build.sh [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
 set -e
 
+BUILD_ATTENTIONS_MODULE="ON"
```
[the remark this answers is not in the document]
Comment (ping1jing2) [reply build.sh:4]: make sense

File: build.sh [hunk 2]
```diff
@@ -175,6 +176,21 @@ function build_memory_saver()
...
 }
 
+function build_attentions_kernels() 
+{
+    CUSTOM_OPP_DIR="${CURRENT_DIR}/python/attentions/attentions"
+    KERNEL_DIR="csrc/attentions/build"
+
+    cd "$KERNEL_DIR" || exit
+
+    echo "run build laser attention kernel"
```
Comment (ping1jing2) [inline build.sh]: please update the comment

Comment (Svoloch2940194) [reply build.sh]: done

File: python/attentions/attentions/plugin/__init__.py
Comment (Svoloch2940194) [reply python/attentions/attentions/plugin/__init__.py:1]: This folder is created automatically by the build_plugin.sh script, so it's not needed. It should contain fully built source libraries.

File: csrc/attentions/build/build.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+# See the Mulan PSL v2 for more details.
+
+set -e
+BUILD_DIR=$(dirname $(readlink -f $0))
+PROJ_ROOT_DIR=${BUILD_DIR}/..
+chmod a-w $BUILD_DIR/*
+
+cd ${PROJ_ROOT_DIR}
+
+echo "\nPROJ_ROOT_DIR = ${PROJ_ROOT_DIR}"
```
[the remark this answers is not in the document]
Comment (Svoloch2940194) [reply csrc/attentions/build/build.sh]: done

File: csrc/attentions/csrc/plugin/find_op_path.cpp [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+std::string ReadLoadPriorityLine(const std::string& configFile)
+{
+    std::ifstream ifs(configFile);
+    std::string line;
+    while (std::getline(ifs, line)) {
+        if (line.find("load_priority=") == 0) {
+            break;
+        }
+    }
+    return line;
```
[the remark this answers is not in the document]
Comment (Svoloch2940194) [reply csrc/attentions/csrc/plugin/find_op_path.cpp:117]: done

File: python/attentions/attentions/cache_agent/cache_agent.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            raise ConfigError(f"The 'blocks_count' in config must > 0, but got {self._config.blocks_count}.")
+        if self._config.steps_count <= 0:
+            raise ConfigError(f"The 'steps_count' in config must > 0, but got {self._config.steps_count}.")
+        
+        if self._config.step_start < 0:
+            raise ConfigError(f"The 'step_start' in config must >= 0, but got {self._config.step_start}.")
+        if self._config.step_interval <= 0:
+            raise ConfigError(f"The 'step_interval' in config must > 0, but got {self._config.step_interval}.")
+        if self._config.block_start < 0:
+            raise ConfigError(f"The 'block_start' in config must >= 0, but got {self._config.block_start}.")
```
[the remark this answers is not in the document]
Comment (Svoloch2940194) [reply python/attentions/attentions/cache_agent/cache_agent.py]: This file is not needed. The folder has been deleted.

**Comment by ping1jing2** [comment]
please resolve the confilct and update your description

File: build.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (ping1jing2) [reply build.sh:4]: please notice this comment

**Comment by ping1jing2** [comment]
please resolve the lint issue and check all the comments

image

File: build.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (Svoloch2940194) [reply build.sh:4]: named by analogy with the DeepEp module
